06-TimeManagement/module_time.c

```diff
@@ -0,0 +1,176 @@
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/device.h>
+#include <linux/err.h>
+#include <linux/uaccess.h>
+#include <linux/kernel.h>
+#include <linux/ctype.h>
+#include <linux/types.h>
+#include <linux/jiffies.h>
+#include <linux/time.h>
+
+
+unsigned long int delta;
+unsigned long int start_module;
+static unsigned long int delay;
+
+static struct timeval abs_time;
+static struct timer_list timer;
+
+
+static void timer_update(void)
+{
+	if (delay)
+		mod_timer(&timer, jiffies + msecs_to_jiffies(delay * HZ));
+	else
+		del_timer(&timer);
```

Comment:
If the delay is less than 1 then the timer will be stopped.
The timer will be turned on again after the delay is set to be greater than or equal to 1.
99	if (tmp) {
100		pr_info("delay: %ld\n", delay);
101		timer_update();

The spam period of the message can only be set to an integer value, because the delay is an integer value. 0.5 s can not be installed, you can from 1c